src/core/moves/mixture/OrderedEventBirthDeathProposal.cpp

```diff
@@ -0,0 +1,316 @@
+ * change too because of the renormalization.
+ * First, some random indices are drawn. Then, the proposal draws a new somplex
+ *   u ~ Uniform(val[index] * alpha)
+ * where alpha is the tuning parameter.The new value is set to u.
+ * The simplex is then renormalized.
+ *
+ * \return The hastings ratio.
+ */
+double OrderedEventBirthDeathProposal::doProposal( void )
+{
+    // get the distribution and make sure it is of the right type
+    AbstractEventTimesDistribution* dist = dynamic_cast<AbstractEventTimesDistribution *>( &event_var->getDistribution() );
+    if (dist == NULL)
+    {
+    	throw RbException("Variable for OrderedEventBirthDeathProposal must inherit from AbstractEventTimesDistribution.");
+    }
+
+    // get the variable
+    OrderedEventTimes &event_variable   = event_var->getValue();
+	const std::set<double>& event_times = event_variable.getEventTimes();
+	size_t num_events                   = event_variable.size();
+
+    // get random number generator
+    RandomNumberGenerator* rng = GLOBAL_RNG;
+
+	// create the hastings ratio
+	double ln_hastings_ratio = 0.0;
+
+	// randomly choose birth or death
+	double u = rng->uniform01();
+
+	if ( u > 0.5 || num_events == 0 )
+	{
+		// pick a birth move
+		was_birth = true;
+
+		// draw the time of the event
+		double new_event_time, ln_prob_time;
+		dist->simulateEventTime(new_event_time, ln_prob_time);
+
+		// add the new time
+		dist->addTime(new_event_time);
+
+		// make sure the dag node is updated
+		event_var->touch();
+
+		// store the chosen time
+		event_time = new_event_time;
+
+		// penalize the drawn time
+		ln_hastings_ratio -= ln_prob_time;
+
+		// for each child, draw a new event
+		const std::vector<DagNode*>& children = event_var->getChildren();
+		for(std::vector<DagNode*>::const_iterator it = children.begin(); it != children.end(); ++it)
+		{
+			if ( (*it)->isStochastic() == true )
+			{
+				AbstractEventsDistribution* child_dist = dynamic_cast<AbstractEventsDistribution *>( &(*it)->getDistribution() );
+				if ( child_dist != NULL )
+				{ // this child is an event distribution
+					ln_hastings_ratio -= child_dist->addEvent(event_time);
+					(*it)->touch();
+				}
+			}
+		}
+
+		// include the hastings ratio for the bias at the boundary
+		if ( num_events == 0 )
+		{
+			ln_hastings_ratio += std::log(0.5);
+		}
+
+	}
+	else
+	{
+		// pick a death move
+		was_birth = false;
+
+		// choose a random event
+		double new_event_time, ln_prob_time;
+		dist->getRandomTime(new_event_time, ln_prob_time);
+
+		// remove the event
+		dist->removeTime(new_event_time);
+		event_var->touch();
+
+		// store the chosen time
+		event_time = new_event_time;
+
+		// penalize the drawn time
+		ln_hastings_ratio += ln_prob_time;
+
+		// for each child, remove the event at the chosen time
+		const std::vector<DagNode*>& children = event_var->getChildren();
+		for(std::vector<DagNode*>::const_iterator it = children.begin(); it != children.end(); ++it)
```

Comment:
maybe `for(auto& child: event_var->getChildren())` and replace `(*it)-`> with `child->`